### PR TITLE
Fix bad error message for scalar subquery

### PR DESCRIPTION
**Pull request by haozhun**
issue #2165

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java [hunk 1]
```diff
@@ -768,6 +769,10 @@ protected Type visitSubqueryExpression(SubqueryExpression node, AnalysisContext
                         descriptor.getVisibleFieldCount());
             }
 
+            if (true) {
+                throw new SemanticException(SemanticErrorCode.NOT_SUPPORTED, node, "Scalar subqueries not yet supported");
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java]: Static import `NOT_SUPPORTED`

**Comment by electrum** [comment]
This implementation was suggestion. There did not seem to be a good way to handle this downstream. Will let @martint review this.

File: presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java [hunk 2]
```diff
@@ -139,6 +139,9 @@ public void testScalarSubQueryException()
             throws Exception
     {
         assertFails(NOT_SUPPORTED, "SELECT 'a', (VALUES (1)) GROUP BY 1");
+        assertFails(NOT_SUPPORTED, "SELECT 'a', (SELECT (1))");
+        analyze("SELECT * FROM (SELECT 1) t1(x) JOIN (SELECT 1) t2(x) ON t1.x=t2.x");
```
Comment (martint) [inline presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java]: This query does not really contain a scalar subquery, so I'd remove it from the test.

Comment (martint) [inline presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java]: Add a test like:

```sql
SELECT * FROM t WHERE (VALUES 1) = 2
```

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java [hunk 3]
```diff
@@ -173,11 +174,35 @@ public ExpressionAnalyzer(Analysis analysis, Session session, Metadata metadata,
...
     }
 
+    private class ScalarSubqueryDetector extends DefaultTraversalVisitor<Void, Void>
+    {
+        @Override
+        protected Void visitInPredicate(InPredicate node, Void context)
+        {
+            Expression valueList = node.getValueList();
+            if (valueList instanceof SubqueryExpression) {
+                super.visitSubqueryExpression((SubqueryExpression) valueList, context);
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java]: This branch needs to call `process(node.getValue(), context)`

**Comment by martint** [comment]
Let me know when you've addressed my comments and I'll merge it.

**Comment by martint** [comment]
Merged

**Pull request closed without merging**